kubernetes/check.py

```diff
@@ -156,6 +162,11 @@ def _perform_kubelet_checks(self, url, instance):
                 self.service_check(service_check_base, AgentCheck.CRITICAL, tags=instance.get('tags', []))
 
     def check(self, instance):
+        # Leader election
+        if self.leader_candidate:
+            self.kubeutil.refresh_leader()
+            self._collect_events = self.kubeutil.is_leader or _is_affirmative(instance.get('collect_events', DEFAULT_COLLECT_EVENTS))
```

Comment:
You still need to re-initialise `self.event_retriever`, or `set_kinds` if already set. But it might be `None` (if not leader not `kube_service` is enabled).
If we lose leadership, we should go back to filtering events.

I think that can be moved to a private method `_configure_events(self, instance, is_leader)`